runners/direct-java/src/main/java/org/apache/beam/runners/direct/TransformExecutor.java

```diff
@@ -138,13 +141,6 @@ private void processElements(
 
         evaluator.processElement(value);
 
-        // Report the physical metrics after each element
-        MetricUpdates deltas = metricsContainer.getUpdates();
-        if (deltas != null) {
-          context.getMetrics().updatePhysical(inputBundle, deltas);
```

Comment:
To do this with the accumulable metrics containers would mean synchronization after each element is processed (as was done in the existing direct metrics implementation) as well as creating a separate metrics container instance for each element.

If reporting attempted counters during processing is a crucial feature for Direct runner we can  return the previous implementation and not use the accumulable version. I think, however, that the reuse of code and reduction of LOC >> this fairly niche feature in the direct runner.